docs/G-Codes.md

```diff
@@ -1313,6 +1313,20 @@ print.
 #### SDCARD_RESET_FILE
 `SDCARD_RESET_FILE`: Unload file and clear SD state.
 
+### [axis_twist_compensation]
+
+The following commands are available when the
+[axis_twist_compensation config
+section](Config_Reference.md#axis_twist_compensation) is enabled.
+
+#### AXIS_TWIST_COMPENSATION_CALIBRATE
+`AXIS_TWIST_COMPENSATION_CALIBRATE [N_POINTS=<value>]`: Initiates the X twist calibration
+wizard. `N_POINTS` specifies the number of points along the X axis to calibrate
+at and defaults to 3.
+
+#### AXIS_TWIST_COMPENSATION_CLEAR
+`AXIS_TWIST_COMPENSATION_CLEAR`: Clears the current compensation values.
```

Comment:
This is the equivalent of BED_MESH_CLEAR, except there are no profiles on this feature.

> how do they get the compensation back afterwards?

Restarting klipper would reload the compensation from the config (unless you used SAVE_CONFIG).

I can remove this command and let the user delete the appropriate section themselves from the klipper config if they want to clear the compensation. Your call :)

Reply:
Correct me if I’m wrong, but would you not want to clear the data before using a paper test to adjust bed screws (not using the probe)? Assuming that the probe is more offset than the nozzle, that is.

Reply:
I don't have bed screws, so I can't run the BED_SCREWS_CALIBRATION myself. But from what I understand, this calibration will only move the nozzle to the 4 corners at a height of 0, and let you adjust the screws to that the bed plane is parallel to the xy plane of the extruder. Does this calibration use the probe at all? If it does not, it won't be affected by this PR.

Reply:
There is BED_SCREWS_ADJUST which is as you say, and also SCREWS_TILT_CALCULATE which does indeed use the probe. In either case, wouldn’t adjusting the screws immediately invalidate the x-twist calibration?

My uninformed instinct is that for printers with severe x twist like mine, you should clear the x-twist calibration, then level the bed using BED_SCREWS_ADJUST, then do the x-twist calibration, in that order. And if you do have a lot of twist, and your probe is strongly offset, don’t use SCREWS_TILT_CALCULATE.

Reply:
Oh that's a nice feature!

Disclaimer: all this is just my understanding of things since I can't test it.

It shouldn't matter if you have twist compensation enabled when doing BED_SCREWS_ADJUST since it will not use the probe. As for SCREWS_TILT_CALCULATE, it's a feature very similar to the bed mesh, but instead of a height map, you get a number of turns to adjust your screws. As such, this SCREWS_TILT_CALCULATE *must* be done after applying the twist compensation, otherwise the result will be tilted, same as the bed mesh.

> In either case, wouldn’t adjusting the screws immediately invalidate the x-twist calibration?

It shouldn't, the twist calibration measures the difference between the height measured from the probe and the real height of the nozzle, same as probe_calibrate, but all along the X axis. Unless your bed was strongly tilted when doing the twist calibration, the introduced error should be negligible.

So maybe this AXIS_TWIST_COMPENSATION_CLEAR is useless after all...